boa/src/builtins/array/mod.rs

```diff
@@ -362,13 +362,10 @@ impl Array {
             arguments = Vec::new();
         }
         // 4.
-        let join_result = _ctx.call(&method, this, &arguments);
-        let match_string = match join_result {
-            Ok(v) => match *v {
-                ValueData::String(ref s) => (*s).clone(),
-                _ => "".to_string(),
-            },
-            Err(v) => format!("error: {}", v),
+        let join = ctx.call(&method, this, &arguments)?;
+        let match_string = match join.data() {
+            ValueData::String(ref s) => s.clone(),
+            _ => "".to_owned(),
         };
```

Comment:
I would change this for an `if let`, and I think we don't need to clone here, right?:
```suggestion
        let match_string = if let ValueData::String(ref s) = join.data() {
            s
        } else {
            ""
        };
```